core/src/test/java/org/elasticsearch/cluster/routing/UnassignedInfoTests.java

```diff
@@ -343,4 +346,20 @@ public void testFindNextDelayedAllocation() {
 
         assertThat(UnassignedInfo.findNextDelayedAllocation(baseTime + delta, clusterState), equalTo(expectMinDelaySettingsNanos - delta));
     }
+
+    public void testAllocationStatusSerialization() throws IOException {
+        assertThat(AllocationStatus.DECIDERS_NO.getValue(), equalTo((byte) 0));
```

Comment:
I wonder if we need that kind of verbosity here. With explicit numerals, we should be protected well enough against BWC issues. it may be sufficient to choose a random AllocationStatus, serialize it and deserialize and check if it's the same.